controllers/tools/applications.js

```diff
@@ -1,22 +1,63 @@
 'use strict';
+const XLSX = require('xlsx');
+
 const applicationService = require('../../services/applications');
 const { purifyUserInput } = require('../../modules/validators');
+const { getFullUrl } = require('../../modules/urls');
 
 function init({ router }) {
-    router.route('/applications/:formId/:applicationId?').get(async (req, res, next) => {
+    // @TODO
+    // domain must be www.biglotteryfund.org.uk or test.blf.digital (eg. CF-protected)
+    // and then WAF rules can protect us
+    router.route('/applications/:formId?/:applicationId?').get(async (req, res, next) => {
         try {
+            let forms, applications, formTitle, applicationData;
             const formId = purifyUserInput(req.params.formId);
 
-            const applications = await applicationService.getApplicationsByForm(formId);
-            if (!applications) {
-                return next();
+            if (!formId) {
+                // fetch all forms instead for a list
+                formTitle = 'All online forms';
+                forms = await applicationService.getAvailableForms();
+            } else {
+                // get applications for a given form
+                applications = await applicationService.getApplicationsByForm(formId);
+                if (!applications) {
+                    return next();
+                }
+
+                formTitle = applications[0].formTitle;
+
+                if (req.query.download) {
+
+                    // build a link to the application page
+                    let urlBase = getFullUrl(req);
+                    urlBase = urlBase.split('?')[0]; // remove querystring
```

Comment:
Minor thing but you can maybe use the bultin `url.parse` to get the part of the URL you need. Don't worry about it for this PR